### DISCUSSION TITLE
plugins: filter in forms.py

### DISCUSSION THREAD

**Discussion by andershagman** [netbox-community/netbox#13119] [Help Wanted!] [answered]
Netbox ver 3.5.4

How do you set a filter in forms.py to only show desired list in DynamicModelChoiceField?
The code shows the whole list as if all() has bin used.

Here is an example ( SiteGroup.objects.filter(name__icontains="L3") ). It works from nbshell.
```python
class L3SiteForm(NetBoxModelForm):
    sitegroup = DynamicModelChoiceField(
        queryset=SiteGroup.objects.filter(name__icontains="L3")
    )
    l3vrf = DynamicModelMultipleChoiceField(
        queryset=VRF.objects.all(),
        required=False,
    )
```

**Comment by kkthxbye-code** [comment]
https://github.com/netbox-community/netbox/blob/bc7678c7160ee1746287b8d8ba21134e01c6c499/netbox/utilities/forms/fields/dynamic.py#L22

Just search the source for `query_params` and there will be plenty of examples. The DynamicModelChoiceField is dynamic (how about that), so setting a static queryset does nothing as the field gets the data from the API client-side. Setting query_params will include what ever set in the API request, so make sure you have whatever you want to filter by implemented in your API.

**Reply by andershagman** [reply, marked as answer]
Here is the solution for me.
```
    sitegroup = DynamicModelChoiceField(
        queryset=SiteGroup.objects.all(),
        query_params={'name__ic': ['L3', 'L2']}
    )
```
